### PR TITLE
Moved relevant decoration expansion into formatters for rlistings fix

### PR DESCRIPTION
**Pull request by Melkiades**
Relevant changes for  https://github.com/insightsengineering/formatters/pull/243. To merge after https://github.com/insightsengineering/formatters/pull/246 

Moved from #813

### CODE REVIEW

File: R/tt_toString.R [hunk 1]
```diff
@@ -296,23 +284,7 @@ setMethod(
...
-        main_title(obj) <- tmp_title_vec[1]
-        subtitles(obj) <- c(tmp_title_vec[-1], .quick_handle_nl(subtitles(obj)))
-      } else {
-        subtitles(obj) <- .quick_handle_nl(subtitles(obj))
-      }
-    }
-
-    # Solve \n in footers
-    main_footer(obj) <- .quick_handle_nl(main_footer(obj))
-    prov_footer(obj) <- .quick_handle_nl(prov_footer(obj))
```
Comment (Melkiades) [inline R/tt_toString.R:312]: relevant modification (this was never meant to happen in matrix_form, but in the `MatrixPrintForm` constructor)

File: tests/testthat/setup-fakedata.R [hunk 2]
```diff
@@ -198,12 +199,13 @@ tt_to_test_newline_chars <- function() {
       split_label = "m\nannaggia\nsda\n",
       label_pos = "visible"
     ) %>%
+    split_cols_by("ARM") %>%
```
Comment (shajoezhu) [inline tests/testthat/setup-fakedata.R:202]: why did we need this? It's just an extra test right?

Comment (Melkiades) [reply tests/testthat/setup-fakedata.R:202]: yes it is an additional test

File: R/tt_toString.R [hunk 3]
```diff
@@ -300,23 +288,7 @@ setMethod(
...
-    # Solve \n in titles
-    if (any(grepl("\n", all_titles(obj)))) {
-      if (any(grepl("\n", main_title(obj)))) {
-        tmp_title_vec <- .quick_handle_nl(main_title(obj))
-        main_title(obj) <- tmp_title_vec[1]
-        subtitles(obj) <- c(tmp_title_vec[-1], .quick_handle_nl(subtitles(obj)))
-      } else {
-        subtitles(obj) <- .quick_handle_nl(subtitles(obj))
-      }
-    }
```
Comment (shajoezhu) [inline R/tt_toString.R:312]: this is now resolved in formatters, right? @Melkiades

**Review by shajoezhu — APPROVED** [review]
lgtm! thanks @Melkiades , nice cleaning up

**Pull request merged**